server/src/test/java/org/elasticsearch/cluster/health/ClusterIndexHealthTests.java

```diff
@@ -147,6 +149,7 @@ protected ClusterIndexHealth mutateInstance(ClusterIndexHealth instance) {
             "relocatingShards",
```

Comment:
This is obviously not your doing, but I'm not a big fan of implementations like these of `mutateInstance`. I prefer implementations like:
https://github.com/elastic/elasticsearch/blob/00f2b9a779adcd52c8172367e6c63132393c8cf9/server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java#L85-L85
because they're much more compact and readable (IMO).

I'll leave it up to you if you want to refactor that in this PR or you keep it like this -- if you even agree, which isn't required of course.

Reply:
I agree it's a much better pattern with the vars and the pattern matching instead of a legacy switch.

I might even go one step further and suggest using a builder defined in an anonymous class inside the test. That would further help contain and simplify the mutation method.

That said, I'm going to open a seperate PR for that I think. This one is already quite large and I'm hesitant to introduce more changes.